sdk/tables/azure-data-tables/azure/data/tables/aio/_table_batch_async.py

```diff
@@ -237,10 +236,8 @@ def update(self, entity: EntityType, mode: Union[str, UpdateMode] = UpdateMode.M
         match_condition = kwargs.pop("match_condition", None)
         etag = kwargs.pop("etag", None)
         if match_condition and not etag:
-            try:
-                etag = entity.metadata.get("etag", None)  # type: ignore[union-attr]
-            except (AttributeError, TypeError):
-                pass
+            if isinstance(entity, TableEntity):
+                etag = entity.metadata.get("etag", None)
```

Comment:
The default is `None by default :)
```suggestion
                etag = entity.metadata.get("etag")
```

Reply:
Updated